frontend/lib/app.dart

```diff
@@ -25,107 +26,127 @@ const homeRouteParamCardIndexName = 'cardIndex';
+            return HomePage(
+              initialTabIndex: int.parse(state.pathParameters[homeRouteParamTabIndexName]!),
+              initialCardIndex: int.parse(state.pathParameters[homeRouteParamCardIndexName]!),
+            );
           },
         ),
+        GoRoute(
+          path: introRouteName,
+          builder: (BuildContext context, GoRouterState state) => const IntroScreen(),
+        ),
       ],
-    ),
-    GoRoute(
-        path: homeRouteName,
-        builder: (BuildContext context, GoRouterState state) {
-          return HomePage();
-        }),
-    GoRoute(
-      path: '$homeRouteName/:$homeRouteParamTabIndexName/:$homeRouteParamCardIndexName',
-      builder: (BuildContext context, GoRouterState state) {
-        return HomePage(
-            initialTabIndex: int.parse(state.pathParameters[homeRouteParamTabIndexName]!),
-            initialCardIndex: int.parse(state.pathParameters[homeRouteParamCardIndexName]!));
-      },
-    ),
-    GoRoute(
-      path: introRouteName,
-      builder: (BuildContext context, GoRouterState state) {
-        return IntroScreen();
-      },
-    ),
-  ],
-);
-
-class App extends StatelessWidget {
-  const App({super.key});
+    );
+  }
 
   @override
   Widget build(BuildContext context) {
-    final settings = Provider.of<SettingsModel>(context);
-    final mapStyleUrl = settings.enableStaging
-        ? buildConfig.mapStyleUrl.staging
-        : isProduction()
-            ? buildConfig.mapStyleUrl.production
-            : isLocal()
-                ? buildConfig.mapStyleUrl.local
-                : buildConfig.mapStyleUrl.showcase;
+    return FutureBuilder(
+      future: _initFuture,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState != ConnectionState.done) {
+          return const SizedBox.shrink();
+        }
 
-    final graphqlUrl = settings.enableStaging
-        ? buildConfig.backendUrl.staging
-        : isProduction()
-            ? buildConfig.backendUrl.production
-            : isLocal()
-                ? buildConfig.backendUrl.local
-                : buildConfig.backendUrl.showcase;
+        final settings = Provider.of<SettingsModel>(context);
+        final router = _createRouter(settings);
 
-    final projectId = isProduction()
-        ? buildConfig.projectId.production
-        : isLocal()
-            ? buildConfig.projectId.local
-            : buildConfig.projectId.showcase;
+        final mapStyleUrl = settings.enableStaging
+            ? buildConfig.mapStyleUrl.staging
+            : isProduction()
+                ? buildConfig.mapStyleUrl.production
+                : isLocal()
+                    ? buildConfig.mapStyleUrl.local
+                    : buildConfig.mapStyleUrl.showcase;
 
-    // Load default language from settings
-    WidgetsBinding.instance.addPostFrameCallback((_) {
-      if (settings.language != null) {
-        LocaleSettings.setLocaleRaw(settings.language!);
```

Comment:
moved to locale_initializer so that the saved locale is initialized at application startup